File: src/main/java/axirassa/overlord/CommandLine.java

package axirassa.overlord;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;

public class CommandLine {

	@Getter
	private String command = null;
	
	@Getter
	private final List<String> arguments = new ArrayList<String>();


	public CommandLine(String command) {
		this.command = command;
	}

	public void addArgument(String argument) {
		arguments.add(argument);
	}


	public void addArguments(List<String> arguments) {
		this.arguments.addAll(arguments);
	}


	public ArrayList<String> buildCommandLine() {
		ArrayList<String> cli = new ArrayList<String>(1 + arguments.size());

		cli.add(command);
		cli.addAll(arguments);

		return cli;
	}

}

File: src/main/java/axirassa/overlord/CommandLine.scala

package axirassa.overlord

import java.util.ArrayList
import java.util.List

class CommandLine(var command : String) {
	val arguments = new ArrayList[String]
	
	def addArgument(argument : String) {
		arguments.add(argument)
	}
	
	def addArguments(arguments: List[String]) {
		arguments.addAll(arguments)
	}
	
	def buildCommandLine = {
		val cli = new ArrayList[String](1 + arguments.size)
		
		cli.add(command)
		cli.addAll(arguments)
		
		cli
	}
}
